src/coreclr/src/debug/debug-pal/unix/diagnosticsipc.cpp

```diff
@@ -121,22 +129,132 @@ IpcStream::DiagnosticsIpc *IpcStream::DiagnosticsIpc::Create(const char *const p
     umask(prev_mask);
 #endif // __APPLE__
 
-    return new IpcStream::DiagnosticsIpc(serverSocket, &serverAddress);
+    return new IpcStream::DiagnosticsIpc(serverSocket, &serverAddress, mode);
 }
 
-IpcStream *IpcStream::DiagnosticsIpc::Accept(ErrorCallback callback) const
+IpcStream *IpcStream::DiagnosticsIpc::Connect(ErrorCallback callback)
+{
+    sockaddr_un clientAddress{};
+    clientAddress.sun_family = AF_UNIX;
+    const int clientSocket = ::socket(AF_UNIX, SOCK_STREAM, 0);
+    if (clientSocket == -1)
+    {
+        if (callback != nullptr)
+            callback(strerror(errno), errno);
+        return nullptr;
+    }
+
+    if (::connect(clientSocket, (struct sockaddr *)_pServerAddress, sizeof(*_pServerAddress)) < 0)
```

Comment:
I was under the impression that `connect` only blocks until the TCP handshake has been completed with the server end and doesn't wait for the server to call `accept`.  We don't do TCP traffic, so I'm not sure there would be any blocking here, but I want to find a more detailed description of this behavior before I say that definitively.  My impression was that `connect` can be called, then return, and I can send data, all before the server calls `accept`.  I'll hunt for more details.

Reply:
If you can't find more detailed and credible info readily I'd suggest confirm the behavior via testing. Write an agent that calls socket() + bind(), and one that calls socket() + bind() + listen(). If neither of those cases cause blocking then you are probably OK. I would also explicit comment that you expect the call not to block so that if we ever discover a case where it does in the future we'll know the behavior wasn't intended.

Reply:
I ran some tests using a modified version of the basic UDS client/server from the `unix(7)` man page.  `connect` is returning immediately.  I'd like to find better documentation on `connect`'s behavior with Unix Domain Sockets, but I'm confident enough to just comment our understanding/expected behaviors for now.